app/livechat/server/lib/Helper.js

```diff
@@ -304,3 +305,17 @@ export const checkServiceStatus = ({ guest, agent }) => {
 
 	return Livechat.online(guest.department);
 };
+
+export const userCanTakeInquiry = (user) => {
+	check(user, Match.ObjectIncluding({
+		status: String,
+		statusLivechat: String,
+	}));
+
+	const { status, statusLivechat } = user;
+	if ((status === 'offline' || statusLivechat !== 'available') && !hasRole(user._id, 'bot')) {
```

Comment:
if you get the field `roles` from user, you can do like this:
```suggestion
	if ((status === 'offline' || statusLivechat !== 'available') && !roles.includes('bot')) {
```

while using `hasRole` is recommended most of the times, in this case you are checking only the `bot` role, and is receiving the user object as a param (that was previously fetched from database).. the `hasRole` would again fetch the same user from database and do an additional find to get the Role.. change to this will prevent then two finds on database.

I think you can change to do this here, but maybe add a `// TODO hasRole` so we can find afterwards if we need to change the `hasRole` function logic

Reply:
Yeah, I agree we can avoid fetching the user from DB again by calling `hasRole` method.
I applied your suggestion, I'm checking the user `roles` properties instead of the `hasRole` function.

Thanks.